packages/flutter_migrate/lib/src/commands/resolve_conflicts.dart

```diff
@@ -0,0 +1,400 @@
+    );
+  }
+
+  final Logger logger;
+
+  final FileSystem fileSystem;
+
+  final Terminal terminal;
+
+  @override
+  final String name = 'resolve-conflicts';
+
+  @override
+  final String description =
+      'Prints the current status of the in progress migration.';
+
+  static const String _conflictStartMarker = '<<<<<<<';
+  static const String _conflictDividerMarker = '=======';
+  static const String _conflictEndMarker = '>>>>>>>';
+
+  @override
+  Future<CommandResult> runCommand() async {
+    final String? projectDirectory = stringArg('project-directory');
+    final FlutterProjectFactory flutterProjectFactory = FlutterProjectFactory();
+    final FlutterProject project = projectDirectory == null
+        ? FlutterProject.current(fileSystem)
+        : flutterProjectFactory
+            .fromDirectory(fileSystem.directory(projectDirectory));
+    final bool isSubcommand = boolArg('flutter-subcommand') ?? false;
+
+    Directory stagingDirectory =
+        project.directory.childDirectory(kDefaultMigrateStagingDirectoryName);
+    final String? customStagingDirectoryPath = stringArg('staging-directory');
+    if (customStagingDirectoryPath != null) {
+      if (fileSystem.path.isAbsolute(customStagingDirectoryPath)) {
+        stagingDirectory = fileSystem.directory(customStagingDirectoryPath);
+      } else {
+        stagingDirectory =
+            project.directory.childDirectory(customStagingDirectoryPath);
+      }
+    }
+    if (!stagingDirectory.existsSync()) {
+      logger
+          .printStatus('No migration in progress. Start a new migration with:');
+      printCommandText('start', logger, standalone: !isSubcommand);
+      return const CommandResult(ExitStatus.fail);
+    }
+
+    final File manifestFile =
+        MigrateManifest.getManifestFileFromDirectory(stagingDirectory);
+    final MigrateManifest manifest = MigrateManifest.fromFile(manifestFile);
+
+    checkAndPrintMigrateStatus(manifest, stagingDirectory, logger: logger);
+
+    final List<String> conflictFiles =
+        manifest.remainingConflictFiles(stagingDirectory);
+
+    terminal.usesTerminalUi = true;
+
+    for (int i = 0; i < conflictFiles.length; i++) {
+      final String localPath = conflictFiles[i];
+      final File file = stagingDirectory.childFile(localPath);
+      final List<String> lines = file.readAsStringSync().split('\n');
+      // We write a newline in the output, this counteracts it.
+      if (lines.last == '') {
+        lines.removeLast();
+      }
+
+      // Find all conflicts
+      final List<Conflict> conflicts = findConflicts(lines, localPath);
+
+      // Prompt developer
+      final CommandResult? promptResult =
+          await promptDeveloperSelectAction(conflicts, lines, localPath);
+      if (promptResult != null) {
+        return promptResult;
+      }
+
+      final bool result =
+          await verifyAndCommit(conflicts, lines, file, localPath);
+      if (!result) {
+        i--;
```

Comment:
I had to re-read to figure out what this was doing; a comment explaining that it's retrying the file would be helpful.